fpm/src/fpm_strings.f90

```diff
@@ -524,10 +524,14 @@ pure function join(str,sep,trm,left,right,start,end) result (string)
          string=string//left_local//str(i)//right_local//sep_local
       endif
    enddo
-   if(trm_local)then
-      string=string//left_local//trim(str(i))//right_local
+   if(size(str).lt.i)then ! if str is zero-sized i will be 1, not zero
```

Comment:
Can't you just check for `size(str) == 0` here? Is there any other situation where this condition would apply anyway?

Reply:
it was time for refactoring that a bit; I think this new version is cleaner still